components/Collaborate/CollabEditor.js

```diff
@@ -85,24 +83,103 @@ const colors = [
 
 const color = colors[Math.floor(Math.random() * colors.length)];
 
+const YDOC_VERSIONS_QUERY = gql`
+  query GetYdocVersions($id: String!) {
+    GetYdoc(id: $id) {
+      versions {
+        createdAt
+        snapshot
+      }
+    }
+  }
+`;
+
+const Editor = ({ provider, currentUser, className, innerRef, onUnmount }) => {
+  useEffect(() => {
+    // console.log('editor mount');
+    return () => {
+      onUnmount();
+    };
+  }, [onUnmount]);
+
+  const ydoc = provider.document;
+  const permanentUserData = new Y.PermanentUserData(ydoc);
+  permanentUserData.setUserMapping(
+    ydoc,
+    ydoc.clientID,
+    JSON.stringify({
+      id: currentUser.id,
+      name: currentUser.name,
+    })
+  );
+
+  const yXmlFragment = ydoc.get('prosemirror', Y.XmlFragment);
+
+  const [state, setState] = useProseMirror({
+    schema,
+    plugins: [
+      ySyncPlugin(yXmlFragment, { permanentUserData }),
+      yCursorPlugin(provider.awareness),
+      yUndoPlugin(),
+      keymap({
+        'Mod-z': undo,
+        'Mod-y': redo,
+        'Mod-Shift-z': redo,
+      }),
+      PlaceholderPlugin(t`Input transcript`),
+    ].concat(exampleSetup({ schema, menuBar: false })),
+  });
+
+  return (
+    <ProseMirror
+      ref={innerRef}
+      state={state}
+      onChange={setState}
+      className={className}
+    />
+  );
+};
+
+/**
+ * @param {Article} props.article
+ */
 const CollabEditor = ({ article }) => {
   const editor = useRef(null);
-  const [editorView, setEditorView] = useState(null);
+  const [showEditor, setShowEditor] = useState(null);
+  const [isSynced, setIsSynced] = useState(false);
   const currentUser = useCurrentUser();
+
+  // onTranscribe setup provider for both Editor and CollabHistory to use.
+  // And, to avoid duplicated connection, provider will be destroyed(close connection) when Editor unmounted.
+  const [provider, setProvider] = useState(null);
+  const { loading, data: getYdocData, refetch } = useQuery(
```

Comment:
That's what it did before, but it hard to calling the `refetch` function from parent (OnDone)

Reply:
I see. I think if we move `useQuery` to [CollabHistory.js](https://github.com/cofacts/rumors-site/pull/545/files/23fb1a5a73cb42e292cf6a30d4e84fc283a60c82#diff-4080c5ad6ccbf9f9d2216ef59b5c83695a1539e1318ef0d4c9c2aa8d236b0c53) we probably won't want to cache it. We can specify `fetchPolicy: 'network-only'` on the `useQuery` in `CollabHistory` so that each time CollabHistory dialog is open, we will always fetch data from network instead of cache. In this case we don't need to refetch on parent (onDone).

Reply:
I added a new commit to fix this and also the no history issue below.